sponsor-dapp/src/components/ContractInteraction.js

```diff
@@ -75,14 +75,22 @@ class ContractInteraction extends Component {
     let withdrawHelper = "";
     let depositHelper = "";
     if (willDefault) {
+      // Round up on the deposit requirement.
       depositHelper = excessMargin
-        ? "Deposit at least " + formatWei(web3.utils.toBN(excessMargin).muln(-1), web3) + marginCurrencyText
+        ? "Deposit at least " +
+          formatWithMaxDecimals(formatWei(web3.utils.toBN(excessMargin).muln(-1), web3), 4, true) +
+          marginCurrencyText
         : "";
     } else {
-      withdrawHelper = excessMargin ? formatWei(excessMargin, web3) + marginCurrencyText + " available" : "";
+      // Round down on the amount available for withdrawal.
+      withdrawHelper = excessMargin
+        ? formatWithMaxDecimals(formatWei(excessMargin, web3), 4, false) + " available"
+        : "";
     }
+
+    // Round down on the number of tokens that will be created.
```

Comment:
I actually think we should round up here. This helper text explains how much money the contract will take out of your account to create the number of tokens you've typed in.

Reply:
Ahhh, I was confused - when reading the variable names, I assumed it meant the number of tokens that would be created. In hindsight, that's what they're entering, so it doesn't make sense. Agreed. Fixed.